packages/shared/src/features/onboarding/hooks/useFunnelTracking.ts

```diff
@@ -190,14 +190,15 @@ export const useFunnelTracking = ({
     const callback = () => {
       if (!isFunnelCompletedRef.current) {
         trackFunnelEventRef.current?.({ name: FunnelEventName.LeaveFunnel });
+        sendBeacon();
```

Comment:
Log context should already handle sending the events via sendBeacon just before the page is closed?

Reply:
@omBratteng we noticed that the leave events API call was being blocked by the browser during page unmount (something similar to [this issue](https://issues.chromium.org/issues/375903219)).

Forcing the beacon to be sent manually resolved the problem

Reply:
But wouldn't this be a more global issue then?
CC: @idoshamun

Reply:
This is the first time we actually send an event on unload and i think there's a race condition here. usually we just send whatever is left in the queue, but here we actually add to the queue. could be after we already sent whatever left

Reply:
Oh ok, why don't we just use the queue here then?

Reply:
we use the queue, but i think that at this point we already ignore the queue with the default implementation. this is why we need to force the analytics to send everything left with a beacon